### PR TITLE
Update Windows font handling and support OpenType fonts

### PR DESCRIPTION
**Pull request by kerrymg**
Made changes to the Windows build so it can properly locate OpenType fonts during font enumeration and loading. Tested font loading with the OpenType version of Source Code Pro and the Noto Sans/Serif CJK CS fonts (on a Windows XP VM - presumably later versions of Windows will work also).

Should fix issue #299 

I also took the liberty to upgrade the Freetype library to the latest version along with some of Freetype's dependencies (zlib and libpng).

### CODE REVIEW

File: garglk/fontwin.c [hunk 1]
```diff
@@ -228,6 +191,62 @@ int CALLBACK propfont(
     return 1;
 }
 
+static void make_font_filepath(const char *filename, char *filepath, size_t length)
+{
+    // create the absolute path to the font file
+    filepath[0] = '\0';
+    if (!strstr(filename, ":") && getenv("SYSTEMROOT"))
```
Comment (cspiegel) [inline garglk/fontwin.c]: As long as this is being touched, it really ought to be updated to be safe against buffer overflows.  There are lots of places in Gargoyle that are (potentially) susceptible to overflows, and they should all be addressed at some point, but at least we can plug some of them now.  I'd recommend something like:

```c
if (!strstr(filename, ":") && getenv("SYSTEMROOT"))
    snprintf(filepath, length, "%s\\Fonts\\%s", getenv("SYSTEMROOT"), filename);
else
    snprintf(filepath, length, "%s", filename);
```

File: garglk/fontwin.c [hunk 2]
```diff
@@ -228,6 +191,62 @@ int CALLBACK propfont(
...
+    char face[256];
+    char filename[256];
+
+    // open the Fonts registry key
+    if (RegOpenKeyEx(HKEY_LOCAL_MACHINE, "Software\\Microsoft\\Windows NT\\CurrentVersion\\Fonts", 0, KEY_QUERY_VALUE, &hkey) != ERROR_SUCCESS)
+        return FALSE;
+
+    // check for a TrueType font
+    strncpy(face, facename, sizeof face);
+    strcat(face, " (TrueType)");
```
Comment (cspiegel) [inline garglk/fontwin.c]: As above, let's take the opportunity to make this safe:

```c
snprintf(face, sizeof face, "%s (TrueType)", facename);
```

File: garglk/fontwin.c [hunk 3]
```diff
@@ -228,6 +191,62 @@ int CALLBACK propfont(
...
+    if (RegQueryValueEx(hkey, face, NULL, NULL, (PBYTE)filename, &size) == ERROR_SUCCESS)
+    {
+        make_font_filepath(filename, filepath, length);
+        RegCloseKey(hkey);
+        return TRUE;
+    }
+
+    // check for an OpenType font
+    strncpy(face, facename, sizeof face);
+    strcat(face, " (OpenType)");
```
Comment (cspiegel) [inline garglk/fontwin.c]: Ditto `snprintf`.

**Comment by kerrymg** [comment]
Here is a screenshot of Gargoyle displaying Chinese characters:

screen shot 2017-12-24 at 12 42 55

(The font is Noto Sans CJK SC)

**Comment by cspiegel** [comment]
Awesome, thank you for tackling this.

**Pull request merged**